File: projects/number-guessing/go/guess.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())
	secret := rand.Intn(100) + 1
	attempts := 0

	fmt.Println("Guess the number (1-100)")

	for {
		var guess int
		fmt.Print("Your guess: ")
		fmt.Scanln(&guess)
		attempts++

		if guess < secret {
			fmt.Println("Too low")
		} else if guess > secret {
			fmt.Println("Too high")
		} else {
			fmt.Printf("Correct! Attempts: %d\n", attempts)
			break
		}
	}
}

File: projects/number-guessing/cpp/guess.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

int main() {
    srand(time(0));
    int secret = rand() % 100 + 1;
    int attempts = 0;
    int guess;

    cout << "Guess the number (1-100)" << endl;

    while (true) {
        cout << "Your guess: ";
        cin >> guess;
        attempts++;

        if (guess < secret) cout << "Too low" << endl;
        else if (guess > secret) cout << "Too high" << endl;
        else {
            cout << "Correct! Attempts: " << attempts << endl;
            break;
        }
    }
    return 0;
}
